### PR TITLE
REFACTOR methods generating video lists

### PR DESCRIPTION
**Pull request by muskie9**

### CODE REVIEW

File: code/VideoGroup.php [hunk 1]
```diff
@@ -37,138 +37,38 @@ class VideoGroup extends Page
...
-        $order = '"SiteTree"."Title" ASC';
-
-        $entries = Video::get()->where($filter)->sort($order);
-
-        $list = new PaginatedList($entries, Controller::curr()->request);
-        $list->setPageLength($limit);
-        return $list;
+        $groupIDS = $this->VideoGroupIDS();
+        $groupIDS[] = $this->ID;
+        return Video::get()->filter('ParentID', $groupIDS)->sort(array('Title' => 'ASC'));
```
Comment (jsirish) [inline code/VideoGroup.php]: Should we sort this by SiteTree order? Would reflect order from CMS.

Comment (muskie9) [inline code/VideoGroup.php]: That might be better, I was going off of the existing sort. I'll update it to be off of the sort column. Maybe we add a config option later to allow for custom sorting.

File: code/VideoGroup.php [hunk 2]
```diff
@@ -37,138 +42,38 @@ class VideoGroup extends Page
...
-                foreach ($children as $group) {
-                    $output->merge($group->ChildGroups($recursive));
-                }
-                return $output;
-            }
-            return null;
-        } else {
-            return DataObject::get('VideoGroup', "\"ParentID\" = '$this->ID'");
-        }
+        return $this->Children()->filter('ClassName', 'VideoGroup');
```
Comment (muskie9) [inline code/VideoGroup.php]: Should we be returning `AllChildren()` here? There may be an inconsistency as `VideoGroupIDs()` returns `AllChildren()`'s ID's.

**Comment by jsirish** [comment]
I don't believe either `ChildGroups()` or `SubGroups()` are being used. If not they can likely be zapped.

**Pull request merged**